src/Task/Shell.php

```diff
@@ -84,10 +86,12 @@ public function run(ContextInterface $context)
      */
     private function runShell(array $scriptArguments)
     {
+        $config = $this->getConfiguration();
         $arguments = $this->processBuilder->createArgumentsForCommand('sh');
         $arguments->addArgumentArray('%s', $scriptArguments);
 
         $process = $this->processBuilder->buildProcess($arguments);
```

Comment:
This `buildProcess()` method here makes sure the timeout is being set to the process builder:

```php
    public function buildProcess(ProcessArgumentsCollection $arguments)
    {
        $builder = SymfonyProcessBuilder::create($arguments->getValues());
        $builder->setTimeout($this->config->getProcessTimeout());
        $process = $builder->getProcess();

        $this->logProcessInVerboseMode($process);
        $this->guardWindowsCmdMaxInputStringLimitation($process);

        return $process;
    }
```